src/App.jsx

```diff
@@ -14,45 +16,102 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORFIELD = {
+  ALPHABET: 'ALPHABET',
+  LENGTH: 'LENGTH',
+};
+
+function getPrepearedGoods(goods, { sortField, isReversed }) {
+  const prepearedGoods = [...goods];
+
+  if (sortField) {
+    prepearedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORFIELD.ALPHABET:
+          return good1.localeCompare(good2);
+
+        case SORFIELD.LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    return prepearedGoods.reverse();
+  }
+
+  return prepearedGoods;
+}
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = getPrepearedGoods(
+    goodsFromServer,
+    { sortField, isReversed },
+  );
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={classNames(
+            'button is-info',
+            {
+              'is-light': sortField !== SORFIELD.ALPHABET,
+            },
+          )}
+          onClick={() => setSortField(SORFIELD.ALPHABET)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={classNames(
+            'button is-success',
+            {
+              'is-light': sortField !== SORFIELD.LENGTH,
+            },
+          )}
+          onClick={() => setSortField(SORFIELD.LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={classNames(
+            'button is-warning',
+            {
+              'is-light': !isReversed,
+            },
+          )}
+          onClick={() => setIsReversed(!isReversed)}
+        >
+          Reverse
+        </button>
+
+        {(sortField || isReversed) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => setSortField('')}
```

Comment:
reset also is reversed state. it will help to fix tests

![chrome-capture-2023-9-8](https://github.com/mate-academy/react_list-of-goods-js/assets/33029864/3f36ceaa-2ff7-48bc-9dff-021e2610473a)